src/vulkan/transfer_buffer.h

```diff
@@ -34,7 +34,9 @@ class TransferBuffer : public Resource {
   TransferBuffer(Device* device, uint32_t size_in_bytes, Format* format);
   ~TransferBuffer() override;
 
-  Result Initialize(const VkBufferUsageFlags usage);
+  TransferBuffer* AsTransferBuffer() override { return this; }
+  void AddUsageFlags(VkBufferUsageFlags flags) { usage_flags_ |= flags; }
```

Comment:
Should we assert that `buffer_` is `VK_NULL_HANDLE`? I.e. you cannot add usage flags after Initialize?

Reply:
Good point. I'll add that